src/bin/languageserver/mod.rs

```diff
@@ -1318,14 +1332,23 @@ impl<'a> Builder<'a> {
 
     /// Traverses namespace to extract information used later by the language server
     /// This includes hover messages, locations where code objects are declared and used
-    fn build(ns: &ast::Namespace) -> (Vec<FileCache>, Definitions, Types, Implementations) {
+    fn build(
+        ns: &ast::Namespace,
+    ) -> (
+        Vec<FileCache>,
+        Definitions,
+        Types,
+        Declarations,
+        Implementations,
+    ) {
```

Comment:
On a quick glance, seems like a weird API. Usually, a `build` function is used on a builder, made available after configuring the builder.

The method doesn't take `self` but it also doesn't return `Self`, instead it returns some other types. Without reading through some example code where this is actually being used, it's very unclear how and when this should be used.

Reply:
The interface has now been changed. 
The `Builder` type now has a `new` method defined which returns a new `Builder`. We don't have any need to configure the `Builder` as of yet, so no provision has been made for that. 
Then the `build` method on `Builder` consumes the `Builder` and returns `(FileCache, GlobalCache)`, which represent the things we have extracted from `Namespace` that are needed by `SolangServer` to service the requests from languageclient later on. The `build` method doesn't return a new `SolangServer` as we need to combine information from the current run of `build` method with information we already had from the previous run of `build` method.